components/calendar/benches/date.rs

```diff
@@ -0,0 +1,165 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+mod fixtures;
+
+use crate::fixtures::structs::DateFixture;
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BenchmarkGroup, Criterion,
+};
+use icu_calendar::{
+    iso::IsoDay, iso::IsoMonth, iso::IsoYear, AsCalendar, Calendar, Date, DateDuration,
+};
+
+fn bench_dates<A: AsCalendar>(dates: Vec<&mut Date<A>>) {
+    for date in dates {
+        // black_box used to avoid compiler optimization.
+        // Arithmetic
+        date.add(DateDuration::new(
+            black_box(1),
+            black_box(2),
+            black_box(3),
+            black_box(4),
+        ));
+
+        // Retrieving vals
+        let _ = black_box(date.year().number);
+        let _ = black_box(date.month().number);
+        let _ = black_box(date.day_of_month().0);
+
+        // Conversion to ISO.
+        let _ = black_box(date.to_iso());
+    }
+}
+
+#[allow(dead_code)]
+fn bench_calendar<C: Clone + Calendar>(
+    group: &mut BenchmarkGroup<WallTime>,
+    name: &str,
+    fxs: &DateFixture,
+    calendar: C,
+    calendar_date_init: impl Fn(i32, u8, u8) -> Date<C>,
+) {
+    group.bench_function(name, |b| {
+        b.iter(|| {
+            for fx in &fxs.0 {
+                // Instantion from int
+                let mut instantiated_date_calendar = calendar_date_init(fx.year, fx.month, fx.day);
+
+                // Conversion from ISO
+                let date_iso = Date::new_iso_date_from_integers(fx.year, fx.month, fx.day).unwrap();
+                let mut converted_date_calendar = Date::new_from_iso(date_iso, calendar.clone());
+
+                bench_dates(vec![
+                    &mut instantiated_date_calendar,
+                    &mut converted_date_calendar,
+                ]);
```

Comment:
Nit (here and elsewhere): don't allocate a `vec!` inside of the bench loop. Use a plain slice or an array instead.

Reply:
Done